### PR TITLE
Fix crash when opening symbology of invalid layer

### LINKED ISSUE #32041: When opening the symbology properties of an invalid layer, QGIS crashes
on master

### PR DESCRIPTION
**Pull request by m-kuhn**
Fix #32041

### CODE REVIEW

File: src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp [hunk 1]
```diff
@@ -535,8 +535,11 @@ QgsCategorizedSymbolRendererWidget::QgsCategorizedSymbolRendererWidget( QgsVecto
   }
 
   mCategorizedSymbol.reset( QgsSymbol::defaultSymbol( mLayer->geometryType() ) );
-  btnChangeCategorizedSymbol->setSymbolType( mCategorizedSymbol->type() );
-  btnChangeCategorizedSymbol->setSymbol( mCategorizedSymbol->clone() );
+  if ( mCategorizedSymbol )
+  {
+    btnChangeCategorizedSymbol->setSymbolType( mCategorizedSymbol->type() );
```
Comment (nyalldawson) [inline src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp]: Hmmmm, we'll still need to call this method somehow. The layer itself will usually have a wkb type available even when it's invalid. Can you use this to set the symbol type when mCategorizedSymbol doesn't exist?

Comment (m-kuhn) [reply src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp]: The symbol is created from the layer type, just one line before and the type is `UnknownGeometry` https://github.com/qgis/QGIS/pull/32043/files/5b12071026be7741e2426124624217cc136572c3#diff-bcfbc8c19bf3c403c220c39a7756db57R537

Comment (nyalldawson) [reply src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp]: Ah, missed that. So the crash must only be happening with an older project, before we started recording the wkb type alongside layers in projects....

Comment (nyalldawson) [reply src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp]: Anyway, +1 to merge. does graduated need the same fix?

Comment (m-kuhn) [reply src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp]: Yes, it's an older project.
And yea, apparently graduated too

Comment (m-kuhn) [reply src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp]: Just fixed whatever crash I could trigger.

**Comment by m-kuhn** [comment]
If a crash doesn't warrant a merge during hard freeze... what does?

**Pull request merged**